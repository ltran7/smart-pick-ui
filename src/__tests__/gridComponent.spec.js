import React from "react";
import Grid from '../components/GridComponent';
import { create } from "react-test-renderer";

describe("Grid component", () => {

  test("Grid of numbers is rendered with correct number of rows and columns", () => {
    const component = create(<Grid numberOfRows={10} numberOfColumns={5} limit={50} selectionLimit={5} selection={[]} onSelection={jest.fn()} type="number"/>);
    expect(component.toJSON()).toMatchSnapshot();

    const grid = component.root.findAllByType('table');
    const row = component.root.findAllByType('tr');
    const limit = component.root.findAllByType('td');
    expect(grid).toHaveLength(1);
    expect(row).toHaveLength(10);
    expect(limit).toHaveLength(50);
    expect(limit.length/row.length).toBeLessThanOrEqual(5); // Check number of columns
  });

  test("Grid of stars is rendered with correct number of rows and columns", () => {
    const component = create(<Grid numberOfRows={3} numberOfColumns={5} limit={12} selectionLimit={5} selection={[]} onSelection={jest.fn()} type="star"/>);
    expect(component.toJSON()).toMatchSnapshot();

    const grid = component.root.findAllByType('table');
    const row = component.root.findAllByType('tr');
    const limit = component.root.findAllByType('td');
    expect(grid).toHaveLength(1);
    expect(row).toHaveLength(3);
    expect(limit).toHaveLength(12);
    expect(limit.length/row.length).toBeLessThanOrEqual(5); // Check number of columns
  });

  test("Grid of numbers has selected numbers", () => {
    var selectedNumbers = [1, 2, 3];
    const component = create(<Grid numberOfRows={10} numberOfColumns={5} limit={50} selectionLimit={5} selection={selectedNumbers} onSelection={jest.fn()} type="number"/>);
    expect(component.toJSON()).toMatchSnapshot();

    const selection = component.root.findAllByType('button').find((b) => b.props.className.includes('ball-selected'));
    expect(selection).toBeDefined();
  });

  test("Grid of stars has selected stars", () => {
    var selectedStars = [1, 2];
    const component = create(<Grid numberOfRows={10} numberOfColumns={5} limit={50} selectionLimit={5} selection={selectedStars} onSelection={jest.fn()} type="stars"/>);
    expect(component.toJSON()).toMatchSnapshot();

    const selection = component.root.findAllByType('button').find((b) => b.props.className.includes('ball-selected'));
    expect(selection).toBeDefined();
  });

});