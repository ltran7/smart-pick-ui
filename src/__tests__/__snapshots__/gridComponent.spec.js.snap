// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component Grid of numbers has selected numbers 1`] = `
<div
  className="col-12"
>
  <table>
    <tbody>
      <tr>
        <td>
          <button
            className="ball-selected"
            id="1-number"
            onClick={[Function]}
          >
            1
          </button>
        </td>
        <td>
          <button
            className="ball-selected"
            id="2-number"
            onClick={[Function]}
          >
            2
          </button>
        </td>
        <td>
          <button
            className="ball-selected"
            id="3-number"
            onClick={[Function]}
          >
            3
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="4-number"
            onClick={[Function]}
          >
            4
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="5-number"
            onClick={[Function]}
          >
            5
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="6-number"
            onClick={[Function]}
          >
            6
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="7-number"
            onClick={[Function]}
          >
            7
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="8-number"
            onClick={[Function]}
          >
            8
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="9-number"
            onClick={[Function]}
          >
            9
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="10-number"
            onClick={[Function]}
          >
            10
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="11-number"
            onClick={[Function]}
          >
            11
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="12-number"
            onClick={[Function]}
          >
            12
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="13-number"
            onClick={[Function]}
          >
            13
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="14-number"
            onClick={[Function]}
          >
            14
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="15-number"
            onClick={[Function]}
          >
            15
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="16-number"
            onClick={[Function]}
          >
            16
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="17-number"
            onClick={[Function]}
          >
            17
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="18-number"
            onClick={[Function]}
          >
            18
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="19-number"
            onClick={[Function]}
          >
            19
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="20-number"
            onClick={[Function]}
          >
            20
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="21-number"
            onClick={[Function]}
          >
            21
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="22-number"
            onClick={[Function]}
          >
            22
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="23-number"
            onClick={[Function]}
          >
            23
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="24-number"
            onClick={[Function]}
          >
            24
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="25-number"
            onClick={[Function]}
          >
            25
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="26-number"
            onClick={[Function]}
          >
            26
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="27-number"
            onClick={[Function]}
          >
            27
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="28-number"
            onClick={[Function]}
          >
            28
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="29-number"
            onClick={[Function]}
          >
            29
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="30-number"
            onClick={[Function]}
          >
            30
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="31-number"
            onClick={[Function]}
          >
            31
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="32-number"
            onClick={[Function]}
          >
            32
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="33-number"
            onClick={[Function]}
          >
            33
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="34-number"
            onClick={[Function]}
          >
            34
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="35-number"
            onClick={[Function]}
          >
            35
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="36-number"
            onClick={[Function]}
          >
            36
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="37-number"
            onClick={[Function]}
          >
            37
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="38-number"
            onClick={[Function]}
          >
            38
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="39-number"
            onClick={[Function]}
          >
            39
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="40-number"
            onClick={[Function]}
          >
            40
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="41-number"
            onClick={[Function]}
          >
            41
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="42-number"
            onClick={[Function]}
          >
            42
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="43-number"
            onClick={[Function]}
          >
            43
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="44-number"
            onClick={[Function]}
          >
            44
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="45-number"
            onClick={[Function]}
          >
            45
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="46-number"
            onClick={[Function]}
          >
            46
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="47-number"
            onClick={[Function]}
          >
            47
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="48-number"
            onClick={[Function]}
          >
            48
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="49-number"
            onClick={[Function]}
          >
            49
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="50-number"
            onClick={[Function]}
          >
            50
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Grid component Grid of numbers is rendered with correct number of rows and columns 1`] = `
<div
  className="col-12"
>
  <table>
    <tbody>
      <tr>
        <td>
          <button
            className="ball"
            id="1-number"
            onClick={[Function]}
          >
            1
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="2-number"
            onClick={[Function]}
          >
            2
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="3-number"
            onClick={[Function]}
          >
            3
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="4-number"
            onClick={[Function]}
          >
            4
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="5-number"
            onClick={[Function]}
          >
            5
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="6-number"
            onClick={[Function]}
          >
            6
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="7-number"
            onClick={[Function]}
          >
            7
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="8-number"
            onClick={[Function]}
          >
            8
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="9-number"
            onClick={[Function]}
          >
            9
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="10-number"
            onClick={[Function]}
          >
            10
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="11-number"
            onClick={[Function]}
          >
            11
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="12-number"
            onClick={[Function]}
          >
            12
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="13-number"
            onClick={[Function]}
          >
            13
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="14-number"
            onClick={[Function]}
          >
            14
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="15-number"
            onClick={[Function]}
          >
            15
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="16-number"
            onClick={[Function]}
          >
            16
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="17-number"
            onClick={[Function]}
          >
            17
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="18-number"
            onClick={[Function]}
          >
            18
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="19-number"
            onClick={[Function]}
          >
            19
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="20-number"
            onClick={[Function]}
          >
            20
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="21-number"
            onClick={[Function]}
          >
            21
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="22-number"
            onClick={[Function]}
          >
            22
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="23-number"
            onClick={[Function]}
          >
            23
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="24-number"
            onClick={[Function]}
          >
            24
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="25-number"
            onClick={[Function]}
          >
            25
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="26-number"
            onClick={[Function]}
          >
            26
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="27-number"
            onClick={[Function]}
          >
            27
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="28-number"
            onClick={[Function]}
          >
            28
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="29-number"
            onClick={[Function]}
          >
            29
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="30-number"
            onClick={[Function]}
          >
            30
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="31-number"
            onClick={[Function]}
          >
            31
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="32-number"
            onClick={[Function]}
          >
            32
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="33-number"
            onClick={[Function]}
          >
            33
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="34-number"
            onClick={[Function]}
          >
            34
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="35-number"
            onClick={[Function]}
          >
            35
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="36-number"
            onClick={[Function]}
          >
            36
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="37-number"
            onClick={[Function]}
          >
            37
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="38-number"
            onClick={[Function]}
          >
            38
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="39-number"
            onClick={[Function]}
          >
            39
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="40-number"
            onClick={[Function]}
          >
            40
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="41-number"
            onClick={[Function]}
          >
            41
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="42-number"
            onClick={[Function]}
          >
            42
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="43-number"
            onClick={[Function]}
          >
            43
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="44-number"
            onClick={[Function]}
          >
            44
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="45-number"
            onClick={[Function]}
          >
            45
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="46-number"
            onClick={[Function]}
          >
            46
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="47-number"
            onClick={[Function]}
          >
            47
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="48-number"
            onClick={[Function]}
          >
            48
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="49-number"
            onClick={[Function]}
          >
            49
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="50-number"
            onClick={[Function]}
          >
            50
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Grid component Grid of stars has selected stars 1`] = `
<div
  className="col-12"
>
  <table>
    <tbody>
      <tr>
        <td>
          <button
            className="ball-selected"
            id="1-stars"
            onClick={[Function]}
          >
            1
          </button>
        </td>
        <td>
          <button
            className="ball-selected"
            id="2-stars"
            onClick={[Function]}
          >
            2
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="3-stars"
            onClick={[Function]}
          >
            3
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="4-stars"
            onClick={[Function]}
          >
            4
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="5-stars"
            onClick={[Function]}
          >
            5
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="6-stars"
            onClick={[Function]}
          >
            6
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="7-stars"
            onClick={[Function]}
          >
            7
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="8-stars"
            onClick={[Function]}
          >
            8
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="9-stars"
            onClick={[Function]}
          >
            9
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="10-stars"
            onClick={[Function]}
          >
            10
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="11-stars"
            onClick={[Function]}
          >
            11
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="12-stars"
            onClick={[Function]}
          >
            12
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="13-stars"
            onClick={[Function]}
          >
            13
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="14-stars"
            onClick={[Function]}
          >
            14
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="15-stars"
            onClick={[Function]}
          >
            15
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="16-stars"
            onClick={[Function]}
          >
            16
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="17-stars"
            onClick={[Function]}
          >
            17
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="18-stars"
            onClick={[Function]}
          >
            18
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="19-stars"
            onClick={[Function]}
          >
            19
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="20-stars"
            onClick={[Function]}
          >
            20
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="21-stars"
            onClick={[Function]}
          >
            21
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="22-stars"
            onClick={[Function]}
          >
            22
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="23-stars"
            onClick={[Function]}
          >
            23
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="24-stars"
            onClick={[Function]}
          >
            24
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="25-stars"
            onClick={[Function]}
          >
            25
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="26-stars"
            onClick={[Function]}
          >
            26
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="27-stars"
            onClick={[Function]}
          >
            27
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="28-stars"
            onClick={[Function]}
          >
            28
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="29-stars"
            onClick={[Function]}
          >
            29
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="30-stars"
            onClick={[Function]}
          >
            30
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="31-stars"
            onClick={[Function]}
          >
            31
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="32-stars"
            onClick={[Function]}
          >
            32
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="33-stars"
            onClick={[Function]}
          >
            33
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="34-stars"
            onClick={[Function]}
          >
            34
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="35-stars"
            onClick={[Function]}
          >
            35
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="36-stars"
            onClick={[Function]}
          >
            36
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="37-stars"
            onClick={[Function]}
          >
            37
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="38-stars"
            onClick={[Function]}
          >
            38
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="39-stars"
            onClick={[Function]}
          >
            39
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="40-stars"
            onClick={[Function]}
          >
            40
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="41-stars"
            onClick={[Function]}
          >
            41
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="42-stars"
            onClick={[Function]}
          >
            42
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="43-stars"
            onClick={[Function]}
          >
            43
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="44-stars"
            onClick={[Function]}
          >
            44
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="45-stars"
            onClick={[Function]}
          >
            45
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="46-stars"
            onClick={[Function]}
          >
            46
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="47-stars"
            onClick={[Function]}
          >
            47
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="48-stars"
            onClick={[Function]}
          >
            48
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="49-stars"
            onClick={[Function]}
          >
            49
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="50-stars"
            onClick={[Function]}
          >
            50
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Grid component Grid of stars is rendered with correct number of rows and columns 1`] = `
<div
  className="col-12"
>
  <table>
    <tbody>
      <tr>
        <td>
          <button
            className="ball"
            id="1-star"
            onClick={[Function]}
          >
            1
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="2-star"
            onClick={[Function]}
          >
            2
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="3-star"
            onClick={[Function]}
          >
            3
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="4-star"
            onClick={[Function]}
          >
            4
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="5-star"
            onClick={[Function]}
          >
            5
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="6-star"
            onClick={[Function]}
          >
            6
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="7-star"
            onClick={[Function]}
          >
            7
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="8-star"
            onClick={[Function]}
          >
            8
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="9-star"
            onClick={[Function]}
          >
            9
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="10-star"
            onClick={[Function]}
          >
            10
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="ball"
            id="11-star"
            onClick={[Function]}
          >
            11
          </button>
        </td>
        <td>
          <button
            className="ball"
            id="12-star"
            onClick={[Function]}
          >
            12
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
